src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -275,6 +275,17 @@ protected function insert_attachment( $request ) {
 		$attachment->post_mime_type = $type;
 		$attachment->guid           = $url;
 
+		// If the title was not set, use the original filename.
+		if ( empty( $attachment->post_title ) && ! empty( $files['file']['name'] ) ) {
+			// Remove the file extension (after the last `.`)
+			$tmp_title = substr( $files['file']['name'], 0, strrpos( $files['file']['name'], '.' ) );
+
+			if ( ! empty( $tmp_title ) ) {
+				$attachment->post_title = $tmp_title;
+			}
```

Comment:
We have the following above 

```
		$name       = wp_basename( $file['file'] );
		$name_parts = pathinfo( $name );
		$name       = trim( substr( $name, 0, -( 1 + strlen( $name_parts['extension'] ) ) ) );
```

Could we use `$name_parts['extension']`?  So something like this. 

```
$tmp_title = trim( substr( $files['file']['name'], 0, -( 1 + strlen( $name_parts['extension'] ) ) ) );
```

Reply:
yea, that probably makes sense, I just didn't know if it was any more efficient, I guess I could test that question

Reply:
@adamsilverstein what was the result of this?